packages/glimmer-compiler/lib/template-compiler.ts

```diff
@@ -366,7 +378,23 @@ function assertValidYield({ hash }): string {
   }
 }
 
-function assertValidHasBlock({ params }): string {
+function assertValidPartial({ params, hash, escaped }) /* : expr */ {
+  if (params && params.length !== 1) {
+    throw new Error(`Partial found with no arguments. You must specify a template name.`);
```

Comment:
Needs to include location information, without it the source of this error will very hard to track down in large apps.